src/convertToObject.js

```diff
@@ -6,7 +6,43 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  const result = {};
+
+  sourceString.split(';').forEach((item) => {
+    const declarations = item
+      .split(':')
+      .map((declaration) => deleteBreak(declaration));
+
+    if ((declarations[0], declarations[1])) {
+      const key = declarations[0];
+      const value = declarations[1];
+
+      result[key] = value;
+    }
+  });
+
+  function deleteBreak(item) {
+    let start = 0;
+    let end = 0;
+    const condition = 'abcdefghijklmnoprstuvwxyz0123456789-#%()';
```

Comment:
it is looks like as hardcoded string and it is possible to avoid it and such long logic in deleteBreak
should be enough trim declaration and then trim key and value.